enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,924 @@
+      * Set `priorityClassName: hypershift-control-plane`.
+    * Use `priorityClassName: hypershift-control-plane` for the driver operator.
+    * Phase 2: add metrics services + TLS keys.
+
+* **aws-ebs-csi-driver-operator**:
+  * Add a new cmdline option for the guest cluster kubeconfig file location.
+    * Parse both kubeconfigs:
+      * One from projected service account, which leads to the management cluster.
+      * Second from the new cmdline option introduced above. This one leads to the guest cluster.
+    * Rework CSI driver Deployment to run only the control plane parts of the CSI driver, i.e.:
+      * Remove `hostNetwork: true`.
+      * Add `controller` as the first command line arg.
+      * Set env. var. `AWS_REGION=us-east2` with the actual guest cluster region. The region can be read from
+        `Infrastructure` CR.
+      * This removes dependency on AWS metadata service and allows us to run the controller parts of the driver outside
+        of AWS.
+    * Rework CSI driver DaemonSet to run only node parts of the CSI driver.
+      * Add `node` as the first command line arg.
+      * This allows us not to give the DaemonSet any AWS credentials.
+  * Only on HyperShift:
+    * When interacting with Kubernetes API, carefully choose the right kubeconfig to watch / create / update objects in
+      the right cluster.
+    * Replace namespaces in the driver Deployment, PDBs and other objects that are created in the management cluster.
+      They must be created in the same namespace as the operator.
+    * Update the CSI driver control-plane Deployment: 
+      * Give it kubeconfig to the guest cluster as a Secret volume.
+      * Pass it Secret with STS creds generated by control-plane-operator.
+      * Add `token-minter` sidecar, which will create STS secrets to the driver.
+      * Set `priorityClassName: hypershift-control-plane`.
+    * Phase 2: add metrics services + TLS keys.
+
+* **AWS EBS CSI driver**:
+  * No changes there!
+    * Controller parts will run in the management cluster, but they get kubeconfig to the guest cluster.
+    * Node DaemonSet will get the default projected kubeconfig that can access objects only in the
+      openshift-cluster-csi-drivers namespace. 
+
+
+### Risks and Mitigations
+
+#### Storage CRs are in the guest cluster
+
+* Since CRs for cluster-storage-operator, aws-ebs-csi-driver-operator and cluster-csi-snapshot-controller-operator are
+  in the guest cluster, a guest cluster user can DOS the whole CSI driver / CSI snapshots by editing these CRs in an
+  endless loop. The operators will redeploy the operands with each CR change and the new operands will need to wait for
+  leader election to expire, while the old operands are already terminated.
+  * We mitigate this issue a bit by overwriting user changes in the operator CRs by Hosted Cluster Config Operator that
+    runs in HyperShift management cluster. Still, an evil user can easily write a loop that updates the CRs in a busy
```

Comment:
Maybe clarify that it's not just any evil user, but an evil admin of the guest cluster.
IMO, it also doesn't seem that bad compared to a traditional OCP cluster: an evil cluster admin could do many more bad things today than an admin limited to the scope of their guest cluster.

Reply:
Added "evil cluster-admin" everywhere (I hope).